pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -56,6 +57,35 @@ func (l *Prometheus) GetStatus() PrometheusStatus {
 	return l.Status
 }
 
+// +kubebuilder:validation:Enum=name;shard
+type AdditionalLabelSelector string
+
+type TopologySpreadConstraint struct {
+	*v1.TopologySpreadConstraint `json:",inline"`
+
+	//+optional
+	// +listType=atomic
+	// Defines what Prometheus Operator managed labels should be added to labelSelector on the topologySpreadConstraint.
+	AdditionalLabelSelectors []AdditionalLabelSelector `json:"additionalLabelSelectors,omitempty"`
```

Comment:
Re-reading the discussion on the issue, I wonder if the operator's API shouldn't expose an enum value rather than a list?

````
  topologySpreadConstraints:
  - maxSkew: 2
    topologyKey: topology.kubernetes.io/zone
    whenUnsatisfiable: DoNotSchedule
    # ByShard => automatically inject label selectors matching all pods in the shard (e.g. name + shard).
    # ByShard => automatically inject label selectors matching all pods of the resource.
   # None => does nothing.
    additionalLabelSelectors: ByShard|ByResource|None
````

But I don't have great ideas for naming the field...

Reply:
Hmmm 🤔 
I guess it might make the implementation and evolution a bit complex, if for some reason we introduce a new managed label which might make sense add to the topology, we have to think and new combinations.

Using a list of enum as it is we left the combination strategy to the user.

Reply:
Is there a use case where you would define the following?

````
spec:
  topologySpreadConstraints:
   - additionalLabelSelectors: [ShardName]
     ...
````

With the current proposed API, it would be valid but from the original issue, it doesn't seem to be useful and almost always want `additionalLabelSelectors: [ShardName, ResourceName]` or `additionalLabelSelectors: [ResourceName]`. But I may be missing something.

Reply:
Actually!
After think a bit more I didn't noticed any usecase.
I just implemented your idea of having ShardNameAndResource and ResourceName only.